<a href="https://colab.research.google.com/github/GiriSriramula/GiriSriramula/blob/main/ML_Assignment_8_Heart_Disease_UCI.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Import required libs

In [ ]:
import numpy as np
import scipy as sp
from matplotlib import pyplot as plt
import pandas as pd
import seaborn as sns
from sklearn import preprocessing
from sklearn.feature_selection import SelectKBest
from sklearn.feature_selection import chi2
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import GridSearchCV
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import precision_score,recall_score,f1_score, classification_report, accuracy_score

In [ ]:
%matplotlib inline
pd.set_option('display.max_columns', None)

Read data

In [ ]:
data = pd.read_csv('heart.csv')

FileNotFoundError: ignored

In [ ]:
data.head(5)

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

Features

1.  age - Age in years
2.   sex - 1: Male, 0: Female
3.   cp - chest pain type
4.   trestbps - resting blood presssure in mmHg 
5.   chol - serum chlestoral in mg/dl
6.   fbs - fasting blood sugar > 120 (1: True 0: False)
7.   restecg -  resting ecg
8.   thalach - max heart reate achived
9.   exang - exercise induced angina (1: yes, 0: no)
10.  oldpeak - ST depression induced by excercise relative to rest
11.  slope - the slope iof the peak exercise ST segment
12.  ca - number if major vesserls (0-3) coloreed by flurosopy
13.  thal: (3: normal, 6: fixed defect , 7: reversable defect) 

In [ ]:
data.shape


NameError: ignored